src/main/java/lotto/domain/Rank.java

```diff
@@ -2,23 +2,22 @@
 
 import lotto.startegy.MatchStrategy;
 
+import java.math.BigDecimal;
 import java.util.stream.Stream;
 
 public enum Rank {
-    NONE(2, 0, (count, match) -> count <= 2),
-    FIVE(3, 5_000, (count, match) -> count == 3),
-    FOUR(4, 50_000, (count, match) -> count == 4),
-    THIRD(5, 1_500_000, (count, match) -> (count == 5 && !match)),
-    SECOND(5, 30_000_000, (count, match) -> (count == 5 && match)),
-    FIRST(6, 2_000_000_000, (count, match) -> count == 6);
-
-    private final int matchCount;
-    private final int prizeMoney;
+    NONE(new WinningCondition(2, BigDecimal.valueOf(0)), (count, match) -> count <= 2),
+    FIVE(new WinningCondition(3, BigDecimal.valueOf(5_000)), (count, match) -> count == 3),
+    FOUR(new WinningCondition(4, BigDecimal.valueOf(50_000)), (count, match) -> count == 4),
+    THIRD(new WinningCondition(5, BigDecimal.valueOf(1_500_000)), (count, match) -> (count == 5 && !match)),
+    SECOND(new WinningCondition(5, BigDecimal.valueOf(30_000_000)), (count, match) -> (count == 5 && match)),
+    FIRST(new WinningCondition(6, BigDecimal.valueOf(2_000_000_000)), (count, match) -> count == 6);
+
+    private final WinningCondition winningCondition;
     private final MatchStrategy expression;
```

Comment:
제가 기대했던 Money 클래스의 사용은 이렇습니다!
WinningCondition은 현재 아무런 책임이 없기 때문에 오버엔지니어링 처럼 보이는 것 같아요 😢 

```suggestion
    NONE(2, new Money(BigDecimal.valueOf(0)), (count, match) -> count <= 2),
    FIVE(3, new Money(BigDecimal.valueOf(5_000)), (count, match) -> count == 3),
    FOUR(4, new Money(BigDecimal.valueOf(50_000)), (count, match) -> count == 4),
    THIRD(5, new Money(BigDecimal.valueOf(1_500_000)), (count, match) -> (count == 5 && !match)),
    SECOND(5, new Money(BigDecimal.valueOf(30_000_000)), (count, match) -> (count == 5 && match)),
    FIRST(6, new Money(BigDecimal.valueOf(2_000_000_000)), (count, match) -> count == 6);

    private final int matchCount;
    private final Money prizeMoney;
    private final MatchStrategy expression;
```